test/gurka/test_gtfs.cc

```diff
@@ -588,48 +747,114 @@ TEST(GtfsExample, MakeTile) {
+  }
+  // test route colors
+  const auto colors = {"0000ff", "ff0000"};
+  for (const auto& c : colors) {
+    EXPECT_TRUE(routes_colors.find(strtol(c, nullptr, 16)) != routes_colors.end());
+  }
+
+  EXPECT_EQ(transit_nodes, 15);
   EXPECT_EQ(uses[Use::kRoad], 10);
-  // NOTE: there are 4 for every station (3 of those) because we connect to both ends of the closest
-  // edge to the station and the connections are bidirectional (as per usual), plus some more because
-  // the second platform has no parent
-  EXPECT_EQ(uses[Use::kTransitConnection], 10);
-  EXPECT_EQ(uses[Use::kPlatformConnection], 6);
-  EXPECT_EQ(uses[Use::kEgressConnection], 6);
+  EXPECT_EQ(uses[Use::kTransitConnection], 18);
+  EXPECT_EQ(uses[Use::kPlatformConnection], 10);
+  EXPECT_EQ(uses[Use::kEgressConnection], 10);
   // TODO: this is the only time in the graph that we dont have opposing directed edges (should fix)
-  EXPECT_EQ(uses[Use::kRail], 2);
+  EXPECT_EQ(uses[Use::kRail], 3);
 }
 
-TEST(GtfsExample, route) {
-  valhalla::Api result0 =
+TEST(GtfsExample, route_trip1) {
+  std::string res_json;
+  valhalla::Api res =
       gurka::do_action(valhalla::Options::route, map, {"A", "F"}, "multimodal",
                        {{"/date_time/type", "1"},
                         {"/date_time/value", "2023-02-27T05:50"},
-                        {"/costing_options/pedestrian/transit_start_end_max_distance", "20000"}});
-  EXPECT_EQ(result0.trip().routes_size(), 1);
+                        {"/costing_options/pedestrian/transit_start_end_max_distance", "20000"}},
+                       {}, &res_json);
+
+  // test the PBF output
+  EXPECT_EQ(res.directions().routes().size(), 1);
+  EXPECT_EQ(res.directions().routes(0).legs().size(), 1);
+
+  const auto& leg = res.directions().routes(0).legs(0);
+  EXPECT_NEAR(leg.summary().time(), 7358.0, 0.001);
+  EXPECT_NEAR(leg.summary().length(), 40.914, 0.001);
+  EXPECT_EQ(leg.maneuver(0).type(), DirectionsLeg_Maneuver_Type_kStart);
+  EXPECT_EQ(leg.maneuver(1).type(), DirectionsLeg_Maneuver_Type_kTransitConnectionStart);
+  EXPECT_EQ(leg.maneuver(2).type(), DirectionsLeg_Maneuver_Type_kTransit);
+  EXPECT_EQ(leg.maneuver(2).transit_type(), valhalla::TransitType::kMetro);
+
+  const auto& transit_info = leg.maneuver(2).transit_info();
+  EXPECT_EQ(transit_info.onestop_id(), f1_name + "_" + r1_id);
+  EXPECT_EQ(transit_info.headsign(), "hello");
+  EXPECT_EQ(transit_info.transit_stops().size(), 3);
+  EXPECT_EQ(transit_info.transit_stops(0).type(), TransitPlatformInfo_Type_kStation);
+  // TODO: JSON & PBF disagree: see TODO at beginning of triplegbuilder_util.h::AddTransitInfo
+  // EXPECT_EQ(transit_info.transit_stops(0).onestop_id(), f1_name + "_" + st1_id);
+  // EXPECT_EQ(transit_info.transit_stops(2).onestop_id(), f1_name + "_" + st3_id +
+  // "_transit_station");
+  EXPECT_EQ(transit_info.transit_stops(0).arrival_date_time(), "");
+  EXPECT_EQ(transit_info.transit_stops(0).departure_date_time(), "2023-02-27T06:59-05:00");
```

Comment:
`2023-02-27T06:59-05:00` this pisses me off! off by 1 minute! it should depart at 7 am sharp..